contracts/extensions/ERC721AOwnersExplicit.sol

```diff
@@ -6,30 +6,37 @@ pragma solidity ^0.8.0;
 import '../ERC721A.sol';
 
 abstract contract ERC721AOwnersExplicit is ERC721A {
-    uint256 public nextOwnerToExplicitlySet = 0;
+    uint256 public nextOwnerToExplicitlySet;
 
     /**
      * @dev Explicitly set `owners` to eliminate loops in future calls of ownerOf().
      */
     function _setOwnersExplicit(uint256 quantity) internal {
-        require(quantity > 0, 'quantity must be nonzero');
-        require(currentIndex > 0, 'no tokens minted yet');
-        require(nextOwnerToExplicitlySet < currentIndex, 'all ownerships have been set');
-
-        uint256 oldNextOwnerToSet = nextOwnerToExplicitlySet;
-        uint256 endIndex = oldNextOwnerToSet + quantity - 1;
-        // Set the end index to be the last token index
-        if (endIndex > currentIndex - 1) {
-            endIndex = currentIndex - 1;
-        }
+        require(quantity != 0, 'quantity must be nonzero');
+        require(currentIndex != 0, 'no tokens minted yet');
+        uint256 _nextOwnerToExplicitlySet = nextOwnerToExplicitlySet;
+        require(_nextOwnerToExplicitlySet < currentIndex, 'all ownerships have been set');
+
+        // Index underflow is impossible.
+        // Counter or index overflow is incredibly unrealistic.
```

Comment:
Please elaborate a bit more on this line (e.g.- "incredibly unrealistic, collection size would have to be ___")